### PR TITLE
Add transactional annotation

### PR DESCRIPTION
**Pull request by benoitdechateauvieux**
Add management for the @Transactional annotation

Includes the EntityManagerHolder: 
If there is an entityManager already started, it will return it.
If no entityManager is found in the context, it will create a new one and close it at the end of the execution of the method.

### CODE REVIEW

File: commons-api/pom.xml [hunk 1]
```diff
@@ -64,5 +64,31 @@
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.aspectj</groupId>
+      <artifactId>aspectjrt</artifactId>
+      <version>1.8.2</version>
```
Comment (tu-vu-duy) [inline commons-api/pom.xml]: Do not set version in here. It need set on maven-depmgt-pom of exo. https://github.com/exoplatform/maven-depmgt-pom

**Comment by tu-vu-duy** [comment]
- Currently, I see all class implements and unit-test related JPA added on commons-api. It is not true. We need move all class implements and unit-test to commons-component-common. 
- commons-api is project only used for interface, abstract, and commons models. It do not contain implements class.
- Change EntityManagerService to interface and add implements on  commons-component-common with class name EntityManagerServiceImpl
- Change AbstractGenericDAO to abstract class.

File: commons-api/src/main/java/org/exoplatform/commons/api/jpa/EntityManagerHolder.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            } finally {
+                if (emStarted && (entityManager!=null) &&
+                    (!entityManager.getTransaction().isActive())) {
+                    LOG.debug("Closing new EntityManager");
+                    service.closeEntityManager();
+                }
+            }
+        }
+    }
+}
```
Comment (tu-vu-duy) [inline commons-api/src/main/java/org/exoplatform/commons/api/jpa/EntityManagerHolder.java]: Format java code not true: Follow here: http://developer.exoplatform.org/docs/ide/eclipse/settings/

File: commons-api/src/main/java/org/exoplatform/commons/api/jpa/EntityManagerHolder.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * the end of the execution of the method.
+ *
+ * @author <a href="bdechateauvieux@exoplatform.org">Benoit de Chateauvieux</a>
+ * @version $Revision$
+ */
+public class EntityManagerHolder {
+    private static final Log LOG = ExoLogger.getLogger(EntityManagerHolder.class);
+
+    private final static EntityManager EM_PROXY =  (EntityManager) Proxy.newProxyInstance(
+            ExoEntityManagerInvocationHandler.class.getClassLoader(),
```
Comment (tu-vu-duy) [inline commons-api/src/main/java/org/exoplatform/commons/api/jpa/EntityManagerHolder.java]: Why is not `EntityManagerHolder.class.getCLassLoader()` ?

**Comment by trongtt** [comment]
merged

**Pull request closed without merging**